service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/GetWorkspaceManagedIdentityForDeleteStep.java

```diff
@@ -0,0 +1,21 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.managedIdentity;
+
+import bio.terra.workspace.service.resource.controlled.flight.delete.DeleteControlledResourceStep;
+import java.util.UUID;
+
+/**
+ * A simple type extension for GetWorkspaceManagedIdentityForDeleteStep, to type as a
+ * DeleteControlledResourceStep. TODO: figure out if this needs functionality from
+ * DeleteControlledAzureResourceStep (TBD, WOR-787), and adjust as needed
+ */
+public class GetWorkspaceManagedIdentityForDeleteStep extends GetWorkspaceManagedIdentityStep
```

Comment:
Same question, this strikes me as awkward -- what's the downside in adding the marker interface to the base class?

Reply:
I went back and forth on this quite a bit.

Now that I'm thinking of the marker interface as simply indicating usage, as opposed to behavior, I think simply adding a brief comment to the `GetWorkspaceManagedIdentityStep`  is probably a better approach.

I think I was thinking too far ahead, and was framing it as the `DeleteControlledAzureResourceStep`.